<a href="https://colab.research.google.com/github/Anikeje2/Data-Science-Journey/blob/main/statistical_hypothesis_testing_and_ab_testing_python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

A pharmaceutical company wants to check patient compliance to a new soft jell tablet so they measure number of doses per week
Standard tablet average compliance = 5 doses/week

In [4]:
import scipy.stats as stats #importing libraries
import numpy as np
from scipy.stats import chi2_contingency


In [5]:
#Assignment 1
weekly_dose= [6, 7, 5, 6, 8, 7, 6, 7, 5, 6,
7, 8, 6, 7, 5, 6, 7, 8, 6, 7,
5, 6, 7, 6, 8, 7, 6, 7, 5, 6,
7, 8, 6, 7, 5, 6, 7, 8, 6, 7]

In [6]:
# H0: mu = 5, H1: mu!=5 my hypotheses
t_stat, p_value = stats.ttest_1samp(
    weekly_dose, popmean=5)

print(f'T-statistic: {t_stat:.4f}') #finding T-statistic
print(f'P-value:     {p_value:.4f}') #finding p-value

if p_value < 0.05:
    print('Reject H0')
else:
    print('Fail to reject H0')

T-statistic: 10.1605
P-value:     0.0000
Reject H0


Since the p-value < 0.05 is less than the level of significance, the null hypothesis is rejected. This indicates that there is a statistically significant difference between the mean weekly dose taken and the hypothesized value of 5 doses per week.

In [7]:
# Assignment 2
Salaries=[52000, 48000, 61000, 45000, 53000,
55000, 40000, 47000, 60000, 51000,
49000, 58000, 43000, 52000, 54000,
50000, 46000, 62000, 39000, 57000,
52000, 48000, 56000, 44000, 53000]
# H0: mu = 55,000, H1: mu!=55,000 my hypotheses

In [11]:
t_stat,p_value = stats.ttest_1samp(Salaries, 55000)
print(f'T-statistic: {t_stat:.4f}') #finding T-statistic
print(f'P-value:     {p_value:.4f}') #finding p-value
if p_value < 0.05:
    print('Reject H0')
else:
    print('Fail to reject H0')


T-statistic: -3.2198
P-value:     0.0037
Reject H0


The Null hypothesis is rejected because p-value< 0.05 There is sufficient statistical evidence to conclude that the average salary of employees is different from $55,000. In this sample, the average salary is significantly lower than the company’s claimed $55,000.

In [8]:
before = [142, 158, 135, 170, 125, 163, 148, 139, 175, 122, 155, 168, 130, 145, 160]
after  = [160, 172, 148, 185, 140, 178, 165, 150, 190, 138, 170, 182, 147, 162, 175]

In [9]:
# H0: no difference   H1: after != before  The hypotheses
t, p = stats.ttest_rel(before, after)
print(f"T-statistic: {t:.4f}")  #finding T-statistic
print(f"P-value:     {p:.4f}")   #finding p-value

if p < 0.05:
    print("Reject H0")
else:
    print("Fail to reject H0")

T-statistic: -33.1618
P-value:     0.0000
Reject H0


Reject H0
The Null hypothesis is rejected because p-value < 0.05.
There is sufficient statistical evidence to conclude that
sales after the campaign are significantly different from before.
In this sample, the average sales significantly increased
after the marketing campaign was launched.

In [10]:
factory_a = [88, 92, 85, 90, 87, 93, 89, 84, 91, 86,
             90, 88, 92, 85, 89, 87, 91, 86, 90, 88]

factory_b = [82, 79, 85, 80, 83, 78, 81, 84, 77, 82,
             80, 83, 79, 81, 78, 84, 80, 82, 79, 83]

# H0: mu_A = mu_B    H1: mu_A != mu_B
t_stat, p_value = stats.ttest_ind(factory_a, factory_b)
print(f"T-statistic: {t_stat:.4f}")
print(f"P-value:     {p_value:.4f}")

if p_value < 0.05:
    print("Reject H0")
else:
    print("Fail to reject H0")

T-statistic: 9.8129
P-value:     0.0000
Reject H0


The Null hypothesis is rejected because p-value < 0.05.
There is sufficient statistical evidence to conclude that
the average widget quality of Factory A and Factory B are different.
In this sample, Factory A produces significantly higher
quality widgets than Factory B.

In [12]:
# H0: Sign-up rate is independent of page design
# H1: Sign-up rate is associated with page design

table = [[45, 455],
         [62, 438]]

chi2, p_value_chi, dof, expected = chi2_contingency(table)

print(f"Chi2-statistic: {chi2:.4f}")
print(f"P-value:        {p_value_chi:.4f}")
print(f"Degrees of freedom: {dof}")

if p_value_chi < 0.05:
    print("Reject H0")
else:
    print("Fail to reject H0")

Chi2-statistic: 2.6792
P-value:        0.1017
Degrees of freedom: 1
Fail to reject H0


The p-value (0.1017) is greater than 0.05, so we fail to reject the null hypothesis. Therefore, there is insufficient evidence to conclude that page design significantly affects sign-up rates.


**Lilian's Comment**

Great work on this assignment, Aminat! Your notebook is very well-structured, and I like how you clearly labeled each assignment, how  you handle the scipy.stats library.


Highlights of your work:Scenario Adaptation:

You did an excellent job applying the one-sample t-test to the pharmaceutical compliance scenario. Your conclusion that the new tablet significantly improved compliance ($p < 0.05$) was exactly right.

Paired T-Test: Your analysis of the marketing campaign was perfect. You correctly identified that a p-value of $1.0$ means the campaign had no statistical effect on sales.

A/B Testing Implementation: Your setup of the contingency table for the Chi-Square test was flawless.

A Note on Interpretation:In your A/B testing conclusion, you correctly stated "Fail to reject H0" because the p-value ($0.1017$) was greater than $0.05$. However, in your written summary, you mentioned "The Null hypothesis... indicates that there is a difference."Correction: Remember that the Null Hypothesis ($H_0$) always represents "no difference" or "status quo." So, when you fail to reject $H_0$, you are concluding that there is no significant difference between the old and new designs.